test-data/unit/check-warnings.test

```diff
@@ -145,8 +145,70 @@ main:4: warning: Returning Any from function declared to return "int"
 
 [case testReturnAnyForNotImplementedInBinaryMagicMethods]
 # flags: --warn-return-any
+from typing import Any
+class A:
+    def __init__(self, x: Any) -> None:
```

Comment:
Actually the tests in this file are specifically for the `--warn-*-any` flags, and this particular test just tests that if a binary magic method returns `NotImplemented` you don't get a warning about returning `Any` (which apparently requires some effort since `NotImplemented` is declared as having type `Any` in typeshed as well as in test-data/unit/fixtures/notimplemented.pyi.

Tests that all binary methods support returning `NotImplemented` (or that other methods don't!) belong somewhere else, and tests that mypy infers the right return type when various forms of branching are present belong elsewhere again.

I trust that the branching logic is very well tested already, so I don't think we need more tests for that. As to where the basic `return NotImplemented` tests should go, I think the best place is probably `check-classes.test`. This is somewhat arbitrary.